### PR TITLE
Moved API tests that had been part of mozilla-services to be part of this repo

### PR DESCRIPTION
**Pull request by chartjes**
@mythmon r?

### CODE REVIEW

**Comment by rehandalal** [comment]
The contents of `CONTRACT-TESTS.md` should probably be moved to the docs.

And a little nit. I prefer `requirements-test.txt` to `test-requirements.txt`. But I'll let @mythmon decide if he cares about that.

File: CONTRACT-TESTS.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# API Contract Tests
+
+It's highly recommended to use [Virtualenv](https://virtualenv.pypa.io/en/latest/)
+in order to have an isolated environment.
+
+From the root directory of the project
+
+1. `virtualenv venv`
+2. `source venv/bin/activate` to turn on the virtualenv
+2. `./venv/bin/pip install -r test-requirements.txt`
```
Comment (mythmon) [inline CONTRACT-TESTS.md]: These docs seem to collide with our normal setup docs. For example, we suggest naming the virtualenv `venv` as well, and the two aren't compatible. It probably also needs to be called out that these need to be Python 2.

File: test-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+configparser
+flake8
+jsonschema
+pytest
+requests
```
Comment (mythmon) [inline test-requirements.txt]: These should at least pin versions, and preferably use hashing to assure we get the same requirements each time.

File: CONTRACT-TESTS.md (hunk 1 above)
Comment (mythmon) [inline CONTRACT-TESTS.md]: Oh wait, can these tests run in Python 3? I assumed, but then I didn't notice any Python2-only libraries in the requirements below. If that's the case, definitely defer to the standard setup docs for setting up an environment.

**Comment by mythmon** [comment]
+1 to everything @rehandalal said.

This has broken the CI build for other tests, since we also use pytest. This PR either needs to make the two pytest setups into one, or modify `circle.yml` to specify which pytest to use.

File: conftest.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Configuration file for running our tests
+import pytest
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--env",
+        action="store",
+        required=False,
```
Comment (mythmon) [inline conftest.py]: This should either be required, or have a default. Having neither produces strange errors like `configparser.NoSectionError: No section: None`.

**Comment by mythmon** [comment]
Besides that `--env` issue, I think this is ready to merge. After that happens, I can get these to run in our automation.

**Comment by chartjes** [comment]
@mythmon r? Added in a default value for --env

**Comment by mythmon** [comment]
WFM

**Pull request merged**